src/utils/automation-helper.js

```diff
@@ -81,90 +85,93 @@ export const parseOutput = (data, translator) => {
 };
 
 // eslint-disable-next-line consistent-return, no-async-promise-executor
-export const automationRunPolling = async (automationId, runId, translator, time = 0) => {
+export const automationRunPolling = async (automationId, runId, translator) => {
   const runningStatuses = ['queued', 'running', 'not started', 'starting'];
-  const pollInterval = 100;
   const defaultMessage = getTranslation(
     translator,
     'Object.ActionButton.Automation.DefaultAutomationMsg',
     'Automation finished'
   );
-  const automationRun = await getAutomationRun(automationId, runId);
-  const { status } = automationRun;
-  if (runningStatuses.includes(status)) {
-    if (time > 10 * 60 * 1000) {
-      return { ok: false, message: getTranslation(translator, 'geo.findLocation.error.timeout', 'Timeout') };
+  let status = 'queued';
+  let automationRun;
+  const endTime = Date.now() + MAX_POLL_TIME;
+  while (runningStatuses.includes(status)) {
```

Comment:
I do not fully understand :) When Date.now() reaches after endTime then the while loop should break, right?

https://github.com/qlik-oss/sn-action-button/blob/f64613c95513ee72648430f32cd0fb22368873f7/src/utils/automation-helper.js#L99

Reply:
There should be no infinite loop here. However it will always wait for 2 seconds, even if the automation run finished faster than that. Depending on how this is being used, 2 seconds can seem like a long time if the automation run only takes a couple of milleseconds.

Other than that, you could try and break out the polling logic into seperate function to make the intent of this logic a bit more obvious.